File: yurt-convention-api/src/main/java/com/github/liulus/yurt/convention/data/ServiceCode.java
package com.github.liulus.yurt.convention.data;

import com.github.liulus.yurt.convention.exception.ServiceErrorException;
import com.github.liulus.yurt.convention.exception.ServiceException;

/**
 * @author liulu
 * @version V1.0
 * @since 2020/11/9
 */
public interface ServiceCode extends StatusCode {

    default ServiceException failure() {
        return new ServiceException(this);
    }

    default ServiceException failure(Throwable cause) {
        return new ServiceException(this, cause);
    }

    default ServiceErrorException error() {
        return new ServiceErrorException(this);
    }

    default ServiceErrorException error(Throwable cause) {
        return new ServiceErrorException(this, cause);
    }


}

File: yurt-convention-api/src/main/kotlin/com/github/liulus/yurt/convention/data/ServiceCode.kt
package com.github.liulus.yurt.convention.data

import com.github.liulus.yurt.convention.exception.ServiceException

/**
 *
 * @author liulu
 * @version V1.0
 * @since 2020/10/31
 */
interface ServiceCode : StatusCode {
    /**
     * 返回服务异常
     * @return Service Exception
     */
    fun failure(): ServiceException {
        return ServiceException(this)
    }

    /**
     * 返回服务异常
     * @param cause 原始异常
     * @return Service Exception
     */
    fun failure(cause: Throwable): ServiceException {
        return ServiceException(this, cause)
    }
}